Makefile

```diff
@@ -139,13 +144,113 @@ help-install:
 #
 ###############################################################
 
+### Deployment ################################################
+#
+# General release steps:
+# --------------------------------------------------------------
+#  (1) commit all files and remove any untracked files
+#      `git status` should show nothing.
+#  (2) `make release-changelog` to print a suggested
+#      update to CHANGELOG.md. Replace "XXX" with your
+#      intended tag.
+#  (3) create an annotated tag (with -a or -s)
+#  (4) `make release-prep` which:
```

Comment:
this generates a "VERSION is dirty"  probably because the changelog has been updated.  should step 2 result in a commit?

Reply:
Sorry, yes. I'll reverse (1) and (2).